src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs

```diff
@@ -643,6 +637,74 @@ public byte[]? this[ReadOnlySpan<byte> key]
         }
     }
 
+    private unsafe byte[]? GetWithIterator(ReadOnlySpan<byte> key, ColumnFamilyHandle? cf, IteratorManager iteratorManager, ReadFlags flags, out bool success)
+    {
+        success = true;
+
+        using IteratorManager.RentWrapper wrapper = iteratorManager.Rent(flags);
+        Iterator iterator = wrapper.Iterator;
+
+        if (iterator.Valid() && TryCloseReadAhead(iterator, key, out byte[]? closeRes))
+        {
+            return closeRes;
+        }
+
+        iterator.Seek(key);
+        if (iterator.Valid() && Bytes.AreEqual(iterator.GetKeySpan(), key))
+        {
+            return iterator.Value();
+        }
+
+        success = false;
+        return null;
+    }
+
+    private unsafe byte[]? Get(ReadOnlySpan<byte> key, ColumnFamilyHandle? cf, ReadFlags flags)
+    {
+        // TODO: update when merged upstream: https://github.com/curiosity-ai/rocksdb-sharp/pull/61
+        // return _db.Get(key, cf, (flags & ReadFlags.HintCacheMiss) != 0 ? _hintCacheMissOptions : _defaultReadOptions);
+
+        nint db = _db.Handle;
+        nint read_options = ((flags & ReadFlags.HintCacheMiss) != 0 ? _hintCacheMissOptions : _defaultReadOptions).Handle;
+        UIntPtr skLength = (UIntPtr)key.Length;
+        IntPtr handle;
+        IntPtr errPtr;
+        fixed (byte* ptr = &MemoryMarshal.GetReference(key))
+        {
+            handle = cf is null
+                        ? Native.Instance.rocksdb_get_pinned(db, read_options, ptr, skLength, out errPtr)
+                        : Native.Instance.rocksdb_get_pinned_cf(db, read_options, cf.Handle, ptr, skLength, out errPtr);
+        }
+
+        if (errPtr != IntPtr.Zero) ThrowRocksDbException(errPtr);
+        if (handle == IntPtr.Zero) return null;
+
+        try
+        {
+            IntPtr valuePtr = Native.Instance.rocksdb_pinnableslice_value(handle, out UIntPtr valueLength);
+            if (valuePtr == IntPtr.Zero)
+            {
+                return null;
+            }
+
+            int length = (int)valueLength;
+            byte[] result = new byte[length];
+            new ReadOnlySpan<byte>((void*)valuePtr, length).CopyTo(new Span<byte>(result));
+            return result;
+        }
+        finally
+        {
+            Native.Instance.rocksdb_pinnableslice_destroy(handle);
```

Comment:
shouldn't it be `valuePtr` and 

https://github.com/NethermindEth/nethermind/pull/8247/files#diff-37e7eb8b529265d0cd0f25ad1f5eed170ca123883a8dc9ba13d1cef5a2409549R684

should be outside of `try`?

Or maybe it should be both?

Reply:
doesn't care about `valuePtr` its `handle` we are releasing